integration-testing/onboard_test.go

```diff
@@ -0,0 +1,185 @@
+	Enabled       bool   `json:"enabled"`
+	Onboarded     bool   `json:"onboarded"`
+}
+
+// ClustersResponse represents the response containing cluster data.
+type ClustersResponse struct {
+	Clusters      []Cluster `json:"clusters"`
+	TotalSize     int       `json:"total_size"`
+	NextPageToken string    `json:"next_page_token"`
+	PrevPageToken string    `json:"prev_page_token"`
+	Page          int       `json:"page"`
+}
+
+// Config holds configuration values needed for the API requests.
+type Config struct {
+	TenantId     string
+	CloudIdKey   string
+	CloudIdValue string
+	UserId       string
+}
+
+// NewHTTPClient returns a reusable HTTP client.
+func NewHTTPClient() *http.Client {
+	return &http.Client{}
+}
+
+// getAuthorizationHeader returns the Basic Authorization header using the provided credentials.
+func getAuthorizationHeader(key, secret string) string {
+	credentials := fmt.Sprintf("%s:%s", key, secret)
+	encodedCredentials := base64.StdEncoding.EncodeToString([]byte(credentials))
+	return fmt.Sprintf("Basic %s", encodedCredentials)
+}
+
+// buildRequest creates a new HTTP request with common headers.
+func buildRequest(method, url string, body io.Reader, config Config) (*http.Request, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", getAuthorizationHeader(config.CloudIdKey, config.CloudIdValue))
+	req.Header.Add("Accept", "*/*")
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-Tenant-Id", config.TenantId)
+	req.Header.Add("X-User-Id", config.UserId)
+
+	return req, nil
+}
+
+// fetchClusters makes an HTTP request to fetch a list of clusters.
+func fetchClusters(config Config) ([]Cluster, error) {
+	baseURL := "https://cloud.illum.io/api/v1/k8s_cluster" // Replace with actual URL
+
+	params := url.Values{}
+	params.Add("max_results", "10")
+	params.Add("sortBy.field", "1")
+	params.Add("sortBy.sort_order", "1")
+	params.Add("filterBy.field", "ILLUMIO_REGION")
+	params.Add("filterBy.value", "aws-us-west-2")
+
+	fullURL := fmt.Sprintf("%s?%s", baseURL, params.Encode())
+
+	req, err := buildRequest("GET", fullURL, nil, config)
+	if err != nil {
+		return nil, err
+	}
+
+	client := NewHTTPClient()
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request failed with status: %v", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var clustersResponse ClustersResponse
+	if err := json.Unmarshal(body, &clustersResponse); err != nil {
+		return nil, err
+	}
+
+	return clustersResponse.Clusters, nil
+}
+
+// offboardCluster sends a request to offboard a cluster by its ID.
+func offboardCluster(config Config, clusterIds []string) error {
+	offBoardURL := "https://cloud.illum.io/api/v1/k8s_cluster/offboard"
+
+	body := map[string][]string{
```

Comment:
It would be more readable to define a `struct`s with a `json:"..."` tag on each field than to create JSON objects by hand.
Like you did for `ClustersResponse` above.